File: JavaScript/389. Find the Difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    var n = t.length;
    var m = s.length;
    sArr = [...s].sort();
    tArr = [...t].sort()
    var i = 0;
    while(i<m){
        if(sArr[i]!=tArr[i]){
            return tArr[i];
        }
        i++;
    }
    return tArr[n-1];
};

File: Java/389. Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int n = t.length();
        int m = s.length();
        char[] sArr = s.toCharArray();
        char[] tArr = t.toCharArray();
        Arrays.sort(sArr);
        Arrays.sort(tArr);
        int i = 0;
        while(i<m){
            if(sArr[i]!=tArr[i]){
                return tArr[i];
            }
            i++;
        }
        return tArr[n-1];
    }
}
